portal-impl/src/com/liferay/portlet/documentlibrary/antivirus/ClamAntivirusScannerImpl.java

```diff
@@ -27,19 +27,19 @@
 
 	@Override
 	public void scan(File file) throws AntivirusScannerException {
-		Process process = null;
+		int exitValue = 0;
 
 		try {
-			ProcessBuilder processBuilder = new ProcessBuilder(
-				"clamscan", "--stdout", "--no-summary", file.getAbsolutePath());
-
-			processBuilder.redirectErrorStream(true);
-
-			process = processBuilder.start();
-
-			process.waitFor();
+			exitValue = scanProcess("clamdscan", file);
+		}
+		catch (InterruptedException | IOException e) {
+			exitValue = -1;
+		}
 
-			int exitValue = process.exitValue();
+		try {
```

Comment:
@IstvanD , @hudakl 

Why do we have to call scanProcess twice??

Thanks!

Reply:
Hi,

First, we call the scanProcess with **clamdscan** parameter in order to use the Clam daemon for the virus check. We only call the scanProcess for a second time with **clamscan** parameter, if the **clamdscan** was unsuccessful.

Cheers,
Laci

Reply:
Got it, now it makes sense

Anyway, to avoid duplicated calls, this double call should only be done when the daemon is not available. 

I've found this (I do not know if this is updated or not)
> /**
   * clamscan return values (documented from man clamscan)
   *  0 : No virus found.
   *  1 : Virus(es) found.
   * 40: Unknown option passed.
   * 50: Database initialization error.
   * 52: Not supported file type.
   * 53: Can't open directory.
   * 54: Can't open file. (ofm)
   * 55: Error reading file. (ofm)
   * 56: Can't stat input file / directory.
   * 57: Can't get absolute path name of current working directory.
   * 58: I/O error, please check your file system.
   * 62: Can't initialize logger.
   * 63: Can't create temporary files/directories (check permissions).
   * 64: Can't write to temporary directory (please specify another one).
   * 70: Can't allocate memory (calloc).
   * 71: Can't allocate memory (malloc).
   */

For example, if we scan using  the daemon ("clamdscan" option) and it returns a 52 error ("Not supported file type") if we repeat the scan with the "clamscan" option it seems that we are going to obtain the same error, so repeat the analysis doesn't seem to be a good option.

Can't we check if we have the daemon running and if it is, do not call the AVirus again if we have code status bigger than 2?? And only call "clamscan" option if the daemon is not running.

Does it make sense? Thanks!

Reply:
Hey @IstvanD and @hudakl 

did you see my last comment?? 

Thanks!

Reply:
Hi Roberto,

I've seen your comment and I'm still examining the possible return values and I need to perform some additional tests.

I also tried to find a summary of the return values, but I couldn't find any. Could you please send me a link to where this is described?

Thank you,
Laci